### ISSUE TITLE
[BUG] Incorrect resource count in left-hand menu

### ISSUE THREAD

**Issue by janeczku** [rancher/dashboard#4418]
Rancher 2.6.1

1. Login as a non-admin user that has not been added as a member to any cluster on the "Cluster Management" section but has access to a Harvester cluster on the "Virtualization Management" section.
2. Navigate to Global Apps -> Cluster Management

Result: 

The resource count badge next to the "Cluster" Menu entry is showing an incorrect number.

<img width="1497" alt="Screenshot 2021-10-19 at 15 36 48" src="https://user-images.githubusercontent.com/3813921/137921963-495cd975-0675-4c48-96e2-23da10fd59d4.png">

**Comment by richard-cox** [comment]
@janeczku Can you provide steps to reproduce this? Other questions...

- Do you have any Harvester Clusters?
- Were there any recently created or deleted clusters?
- Does the count stay out of sync after refreshing the page?
- Were you signed in as an admin, or user with limited roles/memberships?

**Comment by janeczku** [comment]
@richard-cox Correct, the counter next to "Clusters" in the "Cluster Management" page incorrectly counts Harvester clusters that are not displayed on the cluster list page.

**Comment by janeczku** [comment]
I have updated the steps to reproduce.

**Comment by richard-cox** [comment]
Thanks for the update.

I'm closing this as a duplicate of #5650 (but acknowledge that one was later, it just contains more acute info). That one is blocked on another issue https://github.com/rancher/rancher/issues/34905
